BioscoopSysteemAPI/BioscoopSysteemAPI/Controllers/PaymentController.cs

```diff
@@ -1,37 +1,183 @@
-            this.paymentRepository = paymentRepository;
+            _paymentRepository = paymentRepository;
+            _mapper = mapper;
         }
 
+        // GET: api/payments
+        /// <summary>
+        /// Get all payments.
+        /// </summary>
+        /// <response code="200">Succesfully returns a payment object.</response>
+        /// <returns>A list of payment objects.</returns>
+        [ProducesResponseType(StatusCodes.Status200OK)]
         [HttpGet]
-        public async Task<ActionResult<IEnumerable<Payment>>> GetPayments()
+        public async Task<ActionResult<IEnumerable<PaymentReadDTO>>> GetPayments()
         {
             try
             {
-                var payments = await this.paymentRepository.GetPayments();
+                var domainPayments = await _paymentRepository.GetPaymentsAsync();
 
-                if (payments == null)
+                if (domainPayments == null)
                 {
                     return NotFound();
                 }
-                return Ok(payments);
+
+                var dtoPayments = _mapper.Map<List<PaymentReadDTO>>(domainPayments.Value);
+
+                return Ok(dtoPayments);
+
+            }
+            catch (Exception)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "Error retrieving data from the database");
+            }
+        }
+
+        // GET: api/payments/1
+        /// <summary>
+        /// Get a payment by ID.
+        /// </summary>
+        /// <param name="id">Id of the payment.</param>
+        /// <returns>A payment object.</returns>
+        /// <response code="200">Succesfully returns a payment object.</response>
+        /// <response code="404">Error: The object you are looking for is not found.</response>
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [HttpGet("{id}")]
+        public async Task<ActionResult<PaymentReadDTO>> GetPayment(int id)
+        {
+            try
+            {
+                var domainPayment = await _paymentRepository.GetPaymentByIdAsync(id);
+                var dtoPayment = _mapper.Map<PaymentReadDTO>(domainPayment.Value);
+
+                if (dtoPayment == null)
+                {
+                    return NotFound();
+                }
+                return Ok(dtoPayment);
+            }
+            catch (Exception)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "Error retrieving data from the database");
+            }
+        }
+
+        // DELETE: api/payment/1
+        /// <summary>
+        /// Delete an object by Id in the database.
+        /// </summary>
+        /// <param name="id">Id of the object.</param>
+        /// <returns></returns>
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<PaymentDeleteDTO>> DeletePayment(int id)
```

Comment:
De reden dat ik bij payments weinig methoden had is omdat we bijv. een betaling niet willen kunnen verwijderen. Alleen een reservering kan worden geannuleerd zou je denken. Of een betaling word alleen terugbetaald..